File: pystan/weather/weather_stan.py
#coding:utf-8
from pystan import StanModel
import pandas as pd

weather_data = pd.read_csv('~/github_personal/pystan/weather/data/data_weather.csv',names=['temp','wind'])
trend_data = pd.read_csv('~/github_personal/pystan/weather/data/google_trend.csv',names=['trend'])

data_length = len(weather_data)
weather_temperature = weather_data['temp']
weather_wind = weather_data['wind']
google_trend = trend_data['trend']

data = {'N':data_length,
        'Y':google_trend,
        'weather_temperature':weather_temperature,
        'weather_wind':weather_wind}

stanmodel = StanModel(file='/Users/takahiro-nakano/github_personal/pystan/weather/stan_model.stan')
fit_nuts = stanmodel.sampling(data=data, n_jobs=1, iter=100, chains = 4)
print(fit_nuts.summary())
mcmc_sample = fit_nuts.extract()
a = mcmc_sample['a']
b = np.array([i for i in range(len(a))])
fit_nuts.plot()
plt.show()

File: pystan/weather/weather_stan.R
library(rstan)
library(dplyr)

weather_data = read.csv("data_weather.csv", header = F)
trend_data = read.csv("google_trend.csv", header = F)

data_length = weather_data %>% nrow
weather_temperature = weather_data[,2]
weather_wind = weather_data[,3]
google_trend = trend_data[,2]

dat = list(N = data_length,Y = google_trend,
           weather_temperature = weather_temperature,
           weather_wind = weather_wind)

fit = stan(file = 'stan_model.stan', data = dat, iter = 2000, chains = 4)
a = summary(fit)$summary[,"mean"]
plot(a[1:182])
stan_trace(fit, pars = "weather_temperature_par")
stan_trace(fit, pars = "weather_wind_par")

################################################
################################################
################################################
fit2 = stan(file = 'stan_model2.stan', data = dat, iter = 500, chains = 2)

